Add a Rules section to Slack

ericholscher opened the pull request:
This adds rules that are more explicit than our social rules.
It also addresses the Pitching & Brand issues we've had recently.

ericholscher committed:
Add a Rules section to Slack

This adds rules that are more explicit than our social rules.
It also addresses the Pitching & Brand issues we've had recently.

Bradamant3 commented:
It's a good idea, for sure, but I'd like to suggest (and explain) some additional re-organization when I'm done recovering from a four-hour workshop Code of Conduct Incident Workshop (not altogether irrelevant as it happens ...). Tomorrow.

ericholscher committed:
Merge remote-tracking branch 'origin/master' into no-promos

ericholscher commented:
@Bradamant3 Sounds good.

mjang commented:
Adding my comments from #staff here -- while I like this PR:

ATM our Slack rules seem separate from (but supplement) our Code of Conduct.

aka, when we explain an action, our choices might be:
1) The post violated our CoC
2) The post violated our Slack rules (less serious?)

(IF we want a set of Slack rules that go beyond the CoC, I'm OK with that, but IMO it'd be harder to enforce)

mjang commented on docs/slack.rst:
```diff
@@ -53,7 +53,16 @@ Chances are we have a large group of folks in your neck of the woods,
+- **No Direct Pitches** - If you’re coming here *only* to make a sales pitch, please don’t. If you just put a link in the chat without any context or conversation, we will delete it.
+- **No Brands** - If you're going to post, post as yourself, not as a brand. We are a community of individuals and we all exist outside of our day jobs and personal brands.
+- **Use inclusive language** - Our community is multicultural, multiracial, multigendered, and generally diverse. When posting in channels, avoid using words such as "guys" to address a mixed group. Possible alternatives: "folks", "everyone", "documentarians".
+
```
Context: from a [discussion of a previous issue](https://github.com/writethedocs/www/issues/496), from @Bradamant3 

"TL;DR: promoting your own ideas or thoughts for larger discussion is one thing, promoting your commercial product or service is another (the latter is advertising and if we're going to allow it we should charge for it)."

mjang commented on docs/slack.rst:
```diff
@@ -53,7 +53,16 @@ Chances are we have a large group of folks in your neck of the woods,
+
+We have a few hard and fast rules that we ask people to follow on our Slack:
+
+- **No Direct Pitches** - If you’re coming here *only* to make a sales pitch, please don’t. If you just put a link in the chat without any context or conversation, we will delete it.
```
I think this should be split into different rules:

 **No Direct Pitches** - If you’re coming here *only* to make a sales pitch, please don’t. We will delete them.

**No links without context** - If you put a link in the chat without context or explanation, we will delete it.

(I think this better fits the kind of spam that I've worked through, e.g. the miscellaneous "here's a poll" without context, e.g. https://trello.com/c/GgsMBN3q, https://trello.com/c/8XSRmVqs . I'm tempted to add something like "If you add the same post in multiple rooms without explanation, that's also spam.")

mxsasha commented:
> ATM our Slack rules seem separate from (but supplement) our Code of Conduct.
> 
> aka, when we explain an action, our choices might be:
> 
> 1. The post violated our CoC
> 2. The post violated our Slack rules (less serious?)

Yes, this is a fair point. I do think a separation can be useful, as rules like "don't dump links" don't fit too well in a CoC, I think. The social rules have a closer overlap. I don't think one is by definition more serious than the other. However, I do think we should refer more to the CoC, and also in a more visible way. I also think the wording of the rules, also the existing ones, can be improved.

Before I start working on making specific text suggestions though, I'd also love to hear what ideas @Bradamant3 has on how to approach this.

Bradamant3 commented:
@ericholscher can you update again? And then I'll add another commit on top of yours, for discussion. The first issue I'd like to address is the mixing of rules about what you can post (no brands, no unexplained links -- Mike's distinction is the most useful here I think -- with the rule about inclusive language, which isn't the same kind of critter and needs its own elaboration. My plan is for that to be one commit, which we could then merge. And then a couple of fairly fast follows can be some more reorganization, and explanation about violations of Slack rules and CoC (two items, not one -- I think these are important to discuss separately).
Thanks for patience, all, and sorry for the delay.

mjang commented:
> and explanation about violations of Slack rules and CoC (two items, not one -- I think these are important to discuss separately).

If I'm hearing correctly

*Some, but not all, violations of our Slack rules are also violations of our CoC. *

While I'm OK with this, I'd like us to cite an example.

Bradamant3 commented:
Sorry, Mike, I meant that reorganization was a distinct piece of work from discussions of violations of anything.

ericholscher commented:
@Bradamant3 Want to just push your proposed changes to this PR or a new one? I don't really follow what you're asking for, so it would be useful to just see it.

mjang commented:
I'm OK with merging and iterating.... but IIRC, @mxsasha was waiting for @Bradamant3 's input before proposing additional changes:

> Before I start working on making specific text suggestions though, I'd also love to hear what ideas @Bradamant3 has on how to approach this.

I was trying to push this with my references to the CoC (and probably the additional Slack rules)

Bradamant3 committed:
edit mostly for content organization, IA

Changed order of things because I think the social rules and black sheep take precedence. We need the no sales-y content too, but the other stuff seems to me to affect more users. I also thought that putting the inclusive language bit in with the no sales-y content bits diluted the message.
We have more work to do here, but this commit represents my first concerns about the original PR; I agree with @mxsasha and @mjang that we need to be clearer about the CoC also but this is as much as I can manage atm. (With apologies for any typos etc; I don't usually use the GH UI ...)

Bradamant3 commented:
OK, pushed a commit onto Eric's PR -- please take a look at the commit message there (has the GH UI changed in how it handles this? Seems ... different ...)
We have more to do here, but my commit's intended as an interim step, to fix the first things that I saw but by no means everything else in the comments here.

mxsasha committed:
Language tweaks. Slightly more positive, bit less informal, fewer idioms.

mxsasha committed:
More explicit reference to CoC early in the text

mxsasha committed:
Add note about animations

mxsasha commented:
I made a few more tweaks in 9cea836b76365eda793ee57b320a657a3fafde0a and 767688c330e50381ee5624ee725e1151fbec911c - not that many actually. I also made some of the existing language a bit less idiom-heavy.

Slightly unrelated, but as we're editing anyways, I added a note about disabling animations in 115200c5b4d1f98d12f210d156140fb52c1f3604, as flashy animations can be an accessibility problem.

plaindocs commented:
Hey folks, where is this at? Do we wanna merge and iterate? Or are people planning in picking it up again on writing day? :-p

Bradamant3 commented:
I vote for merge and iterate. atm I can't imagine having bandwidth to pick it up on writing day, and I'm not sure I see the value add there. If @mxsasha will be in Portland maybe?

mjang commented:
There are some comments in the latest iteration that I disagree with, and
commented on before I saw Jennifer's response. Hopefully we can discuss in
person in a couple of weeks.

On Mon, May 6, 2019 at 7:18 AM Jennifer Rondeau <notifications@github.com>
wrote:

> I vote for merge and iterate. atm I can't imagine having bandwidth to pick
> it up on writing day, and I'm not sure I see the value add there. If
> @mxsasha <https://github.com/mxsasha> will be in Portland maybe?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/writethedocs/www/pull/840#issuecomment-489637771>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAZCXKGCG4Q45YC6IAH5XILPUA44VANCNFSM4HABN6FQ>
> .
>


-- 
[image: ForgeRock] <https://www.forgerock.com/> *Mike Jang*
Senior Staff Technical Writer  |  ForgeRock
(503) 381-7368  |  michael.jang@forgerock.com
<firstname.lastname@forgerock.com>
*twitter* @theMikeJang  |  *web* www.forgerock.com

mjang commented:
So yes, +1 for merge and iterate.

On Mon, May 6, 2019 at 7:25 AM Mike Jang <michael.jang@forgerock.com> wrote:

> There are some comments in the latest iteration that I disagree with, and
> commented on before I saw Jennifer's response. Hopefully we can discuss in
> person in a couple of weeks.
>
> On Mon, May 6, 2019 at 7:18 AM Jennifer Rondeau <notifications@github.com>
> wrote:
>
>> I vote for merge and iterate. atm I can't imagine having bandwidth to
>> pick it up on writing day, and I'm not sure I see the value add there. If
>> @mxsasha <https://github.com/mxsasha> will be in Portland maybe?
>>
>> —
>> You are receiving this because you commented.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/writethedocs/www/pull/840#issuecomment-489637771>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AAZCXKGCG4Q45YC6IAH5XILPUA44VANCNFSM4HABN6FQ>
>> .
>>
>
>
> --
> [image: ForgeRock] <https://www.forgerock.com/> *Mike Jang*
> Senior Staff Technical Writer  |  ForgeRock
> (503) 381-7368  |  michael.jang@forgerock.com
> <firstname.lastname@forgerock.com>
> *twitter* @theMikeJang  |  *web* www.forgerock.com
>


-- 
[image: ForgeRock] <https://www.forgerock.com/> *Mike Jang*
Senior Staff Technical Writer  |  ForgeRock
(503) 381-7368  |  michael.jang@forgerock.com
<firstname.lastname@forgerock.com>
*twitter* @theMikeJang  |  *web* www.forgerock.com

mxsasha commented:
I will not be in Portland. But, in favour of merge and iterate.

mxsasha reviewed (approved):
As discussed, merge now, iterate later.

mxsasha commented:
Regarding folks: I don't think it's a bad word to use, but excessive use can make a text excessively informal I think. E.g. "a bunch of interesting folks" doesn't seem like a great way to describe our slack anymore, I feel like it's underrating it.

plaindocs commented on docs/slack.rst:
```diff
@@ -100,10 +105,19 @@ If you see people not following our etiquette guidelines,
 
 * Message them privately, and explain why what they said might have been hurtful to someone, referring to this guide.
-* Ask a :ref:`moderation-team` member privately to talk to the user, include a link to the conversation if possible.
+* Ask a :ref:`moderation-team` member address the situation, include a link to the conversation if possible.
```
```suggestion
* Ask a :ref:`moderation-team` member to address the situation, include a link to the conversation if possible.
```

plaindocs committed:
Update docs/slack.rst

plaindocs commented:
Hey folks, merging this in.

On the "folks" issue, I tend to agree that we've progressed past 'a bunch of folks' when referring to ourselves in the slightly more formal setting, and I will continue to use folks, for example above. Absolutely no need to lint for it.

plaindocs committed:
Merge branch 'master' into no-promos

plaindocs committed:
Merge branch 'master' into no-promos

ericholscher commented:
We should also publicly communicate that we've updated these, not just silently update them.
